insights_dart/lib/src/filter_event_tracker_adapter.dart

```diff
@@ -0,0 +1,68 @@
+import 'event_tracker.dart';
+import 'filter_event_tracker.dart';
+
+/// Wrapper for an EventTracker with associated indexName
+class FilterEventTrackerAdapter implements FilterEventTracker {
+  /// Underlying EventTracker instance.
+  EventTracker tracker;
+
+  /// Name of the index to associate events with.
+  String indexName;
+
+  @override
+  bool isEnabled;
+
+  FilterEventTrackerAdapter(
+    this.tracker,
+    this.indexName, {
```

Comment:
takes a number of default values, indexName, attribute, not focused to filters only, then it could be used for both HitsSearcher internally, and FacetList, without needing extension or ever needing to pass `indexName` or `attribute`

Reply:
otherwise two different adapters, one with just indexName for HitsEventTrackerAdapeter, one with attribute and indexName in constructor FilterEventTrackerAdapter